lib/serviceClient.ts

```diff
@@ -177,6 +177,13 @@ export class ServiceClient {
 
     let result: Promise<HttpOperationResponse>;
     try {
+      if (operationSpec.baseUrl == undefined) {
+        operationSpec.baseUrl = (this as any).baseUri;
```

Comment:
any reason baseUri can't also be an optional property on ServiceClient?

Reply:
That's a good suggestion. I'll do that now.

Reply:
Speaking of base uris have we ensured that we process x-ms-parameterized-host extension correctly in the new operation spec way of building the request?

Reply:
Yes, we definitely do.

Reply:
Fixed in https://github.com/Azure/ms-rest-js/pull/153/commits/8be6cd370d3075c9c33c878912d2b5d94235acd3.